src/css.js

```diff
@@ -378,7 +378,11 @@ jQuery.each( [ "height", "width" ], function( i, dimension ) {
 		set: function( elem, value, extra ) {
 			var matches,
 				styles = getStyles( elem ),
-				isBorderBox = jQuery.css( elem, "boxSizing", false, styles ) === "border-box",
+				scrollBoxSize = support.scrollboxSize() === styles.position,
+
+				// Only fetch boxSizing if required to avoid forcing a reflow (gh-3991)
+				isBorderBox = ( extra || scrollBoxSize ) &&
+					jQuery.css( elem, "boxSizing", false, styles ) === "border-box",
```

Comment:
I like what the code is doing, but the reasons could be a little more clear. Suggestion to improve that (and reduce size a little bit as well):
```patch
diff --git i/src/css.js w/src/css.js
index fc000be..d7181f4 100644
--- i/src/css.js
+++ w/src/css.js
@@ -380,8 +380,9 @@ jQuery.each( [ "height", "width" ], function( i, dimension ) {
 				styles = getStyles( elem ),
 				scrollBoxSize = support.scrollboxSize() === styles.position,
 
-				// Only fetch boxSizing if required to avoid forcing a reflow (gh-3991)
-				isBorderBox = ( extra || scrollBoxSize ) &&
+				// To avoid forcing a reflow, only fetch boxSizing if we need it (gh-3991)
+				boxSizingNeeded = scrollBoxSize || extra,
+				isBorderBox = boxSizingNeeded &&
 					jQuery.css( elem, "boxSizing", false, styles ) === "border-box",
 				subtract = extra && boxModelAdjustment(
 					elem,
@@ -393,7 +394,7 @@ jQuery.each( [ "height", "width" ], function( i, dimension ) {
 
 			// Account for unreliable border-box dimensions by comparing offset* to computed and
 			// faking a content-box to get border and padding (gh-3699)
-			if ( scrollBoxSize && isBorderBox ) {
+			if ( isBorderBox && scrollBoxSize ) {
 				subtract -= Math.ceil(
 					elem[ "offset" + dimension[ 0 ].toUpperCase() + dimension.slice( 1 ) ] -
 					parseFloat( styles[ dimension ] ) -
```

Reply:
That reduces size? Interesting, I thought an extra var would do the opposite...

I agree that is clearer 👍

Reply:
Adding the var and changing the comment:
```
   +38    +11 dist/jquery.js
     =      = dist/jquery.min.js
```
Then changing the order in the `if`:
```
   raw     gz Compared to last run
     =     -1 dist/jquery.js
     =      = dist/jquery.min.js
```
So it didn't really change the size at all (but does make it more readable, so :+1:)... @gibson042 is that the results you were getting?

Reply:
I don't remember exactly, but this looks good to me.